### PR TITLE
Enable errorprone and spotbugs for java 17

### PR DESCRIPTION
**Pull request by olamy**
- add compiler flag to support errorprone for jdk17 and restore spotbugs parsing
- spotbugs is fine with jdk17

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -12,7 +12,9 @@ pipeline {
           steps {
             container('jetty-build') {
               timeout( time: 120, unit: 'MINUTES' ) {
-                mavenBuild( "jdk17", "clean install", "maven3")
+                // configuration needed for errorprone but not for normal build
+                sh "mv .mvn/.jvm.config .mvn/jvm.config"
```
Comment (joakime) [inline Jenkinsfile]: Is there any harm in leaving this `.mvn/jvm.config` configuration enabled for both builds?

Comment (olamy) [reply Jenkinsfile]: nope no issue with jdk11.
maybe to ease this for every build even local? (i.e just having a file ` .mvn/jvm.config`)
look at the content this doesn;t affect anything else than com.sun.tools.javac modules
wdyt?

Comment (joakime) [reply Jenkinsfile]: Yes, lets leave it enabled for any JVM. (even JDK 18 and JDK 19)

Comment (olamy) [reply Jenkinsfile]: fixed

Comment (joakime) [reply Jenkinsfile]: Going to wait for a green build now ... :-)

**Review by joakime — APPROVED** [review]
Build is green.
Lets merge this.

**Pull request merged**